addons/website/static/src/snippets/s_website_form/options.js

```diff
@@ -242,6 +246,9 @@ const FieldEditor = FormEditor.extend({
         const labelText = this.$target.find('.s_website_form_label_content').text();
         if (this._isFieldCustom()) {
             field = this._getCustomField(this.$target[0].dataset.type, labelText);
+            if (this.$target[0].classList.contains("s_website_form_model_required")) {
+                field.controllerRequired = true;
+            }
```

Comment:
I don't understand this diff, I think the idea here is to ensure the field state consistency, but there is already a code in `_setActiveProperties()` that will force the field to be `modelRequired` anyway :thinking:
```javascript
field.modelRequired = classList.contains('s_website_form_model_required');
```

Reply:
Without this, after save, when re-selecting the field, the `controllerRequired` information is lost.

Reply:
Yes, but don't you find it very confusing :thinking:  ? We use the same class
`"s_website_form_model_required"` for `modelRequired` and `controllerRequired` which means every `modelRequired` custom field will be `controllerRequired` and every `controllerRequired` field will be `modelRequired`… So, in the end, isn't equivalent to setting `modelRequired = true` on the field (or a global attribute to say that the field is required in the backend) ?

Reply:
Now using dedicated class [(diff)](https://github.com/odoo/odoo/compare/5b7af3730f04f7c77c60a87dbfa633c43db1069e..abf6ca6b29ba7e86ce9caaaceb624f971a396c72)